adsApp/src/epics-ads/Variable.cpp

```diff
@@ -87,6 +87,19 @@ int ADSVariable::read_from_buffer(const uint32_t size, char *buffer) {
         return EPICSADS_NO_DATA;
     }
 
+    // reset readback flag if read value is the same as written value
+    if (addr->get_operation() == Operation::ReadWrite && await_written_readback) {
+        std::vector<uint8_t> read_buffer(bytes_to_read, 0);
+        memcpy(read_buffer.data(), buffer, bytes_to_read);
+        if (read_buffer == last_written) {
+            await_written_readback = false;
+        }
+        else {
+            // return the written value
+            memcpy(buffer, last_written.data(), last_written.size());
+        }
+    }
```

Comment:
This looks like the goal is to prevent the record's value from jumping between "new value"→"old value"→"new value" when the setpoint is adjusted while a background read is in progress. It this the case?

The current approach has a race condition: if EPICS and the PLC try to change the value at the same time, and the change from EPICS happens just before the other change, that other change will immediately overwrite the value coming from EPICS. The driver expects the value from EPICS to be read back eventually, but this may never happen.

I think this could be avoided by *always* resetting the  `await_written_readback` flag. This ensures that at most one periodic read is affected. If that read already returns the newly set value, great. If not, hopefully the next read will. But, if the next read still does not return the new value, then something else happened in between, and we need to show that.

Reply:
Yes, the purpose here is to avoid jumping of the written value. I removed the value comparison and changed it to always reset the `await_written_readback` flag like you suggested, and it seems to work fine.